### PR TITLE
Feature/segun/get user order history

### PR DESCRIPTION
**Pull request by Segun-Ade**
#### What does this PR achieve?
Get list of all orders for a particular user
#### Screenshots:
DeepinScreenshot_select-area_20200113220750

### CODE REVIEW

File: src/components/userDashboard/components/orders/components/previousOrders.jsx [hunk 1]
```diff
@@ -1,81 +1,82 @@
...
+const PreviousServices = () => {
+    const [tableData, setTableData] = useState('');
+    const data = useSelector(state => state);
+    const {
+        orders: { orderRecord },
+    } = data;
+
+    useEffect(() => {
+        if (orderRecord !== null) {
+            Object.keys(orderRecord).map(key => {
```
Comment (Way2nnadi) [inline src/components/userDashboard/components/orders/components/previousOrders.jsx]: only use `.map` if you want to return something, else use `forEach`

File: src/components/userDashboard/components/orders/components/previousOrders.jsx [hunk 2]
```diff
@@ -1,81 +1,82 @@
...
+                } = orderRecord[key];
+                const tab = {
+                    createdAt: moment(createdAt).format('l'),
+                    displayName,
+                    key: orderID,
+                    orderStatus,
+                };
+                setTableData(tableData => [...tableData, tab]);
+                return orderRecord;
+            });
```
Comment (Way2nnadi) [inline src/components/userDashboard/components/orders/components/previousOrders.jsx]: this is wrong, do not do this

File: src/components/userDashboard/components/orders/components/previousOrders.jsx [hunk 3]
```diff
@@ -1,81 +1,82 @@
...
+    useEffect(() => {
+        dispatch(fetchOrders());
+    }, []);
+
+    return (
+        <div>
+            <Col xs={22} md={24} className="previous-order">
+                {PreviousServices()}
+            </Col>
+        </div>
```
Comment (Way2nnadi) [inline src/components/userDashboard/components/orders/components/previousOrders.jsx]: remove this div

File: src/components/userDashboard/components/orders/components/previousOrders.jsx [hunk 4]
```diff
@@ -1,81 +1,82 @@
...
+                    createdAt,
+                    userInfo: { displayName },
+                } = orderRecord[key];
+                const tab = {
+                    createdAt: moment(createdAt).format('l'),
+                    displayName,
+                    key: orderID,
+                    orderStatus,
+                };
+                setTableData(tableData => [...tableData, tab]);
```
Comment (Way2nnadi) [inline src/components/userDashboard/components/orders/components/previousOrders.jsx]: also do not do this, each time you call set state the component rerenders, you want to only call this once

File: src/components/userDashboard/components/orders/sagas.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                authorization: `Bearer ${token}`,
+            },
+        };
+        const request = yield axios.post(ORDER_API, { userType }, config);
+        if (request.status === 200) {
+            const {
+                data: { data },
+            } = request;
+            yield put(fetchOrdersSuccess(data));
+        }
```
Comment (Way2nnadi) [inline src/components/userDashboard/components/orders/sagas.js]: what happens if it is not `200`

File: src/components/userDashboard/components/orders/components/previousOrders.jsx (hunk 2 above)
Comment (Segun-Ade) [reply src/components/userDashboard/components/orders/components/previousOrders.jsx]: Ok

File: src/components/userDashboard/components/orders/sagas.js (hunk 5 above)
Comment (Segun-Ade) [reply src/components/userDashboard/components/orders/sagas.js]: I'll return the response.statusText in an else block

File: src/components/userDashboard/components/orders/components/previousOrders.jsx (hunk 4 above)
Comment (Segun-Ade) [reply src/components/userDashboard/components/orders/components/previousOrders.jsx]: This is how I am able to change the state of tableData and replace it's previous state with the value of tab

Comment (edbella) [reply src/components/userDashboard/components/orders/components/previousOrders.jsx]: I see what you're trying to achieve. Here's another way. In your prevOrderColumns, use the dataIndex to reach into the data you need. And then for createdAt, use the render method I showed you to reformat the date.

So dataIndex for displayName would be 'userInfo.displayName'.

So solve the "key" issue for the table, set a prop of "rowKey" on the table and pass in any field that is unique in your already defined columns.

Comment (Segun-Ade) [reply src/components/userDashboard/components/orders/components/previousOrders.jsx]: Thanks @edbella , I've fixed it using your suggestions

**Pull request merged**